odoo/addons/base/models/ir_cron.py

```diff
@@ -159,24 +160,37 @@ def _process_jobs(db_name, soft_limit=None):
             with db.cursor() as cron_cr:
                 cls = IrCron
                 cls._check_version(cron_cr)
-                jobs = cls._get_all_ready_jobs(cron_cr)
+                jobs = deque(cls._get_all_ready_jobs(cron_cr))
                 if not jobs:
                     return
                 cls._check_modules_state(cron_cr, jobs)
 
-                for index, job_id in enumerate(job['id'] for job in jobs):
+                locked_jobs = set()
+                while True:
+                    if not jobs:
+                        # we have tried once all ready jobs, were some retriggered?
+                        jobs.extend(cls._get_all_ready_jobs(cron_cr))
+                        if not jobs or all(job['id'] in locked_jobs for job in jobs):
```

Comment:
Code not easy to understand. Need refactor for `locked_jobs`.

Reply:
@Julien00859  I tried to make the code cleaner. However, it is needed to ensure we don't have endless loops.
Maybe I could refactor and move the `while` loop in a separate function that could be tested. What do you think?